src/mozanalysis/experiment.py

```diff
@@ -0,0 +1,509 @@
+                _Don't_ use `experiments`; as of 2019/04/02 it drops data
+                collected after people self-unenroll, so unenrolling users
+                will appear to churn. Must have at least the following
+                columns:
+                    - client_id (str)
+                    - submission_date_s3 (str)
+                    - data columns referred to in `metric_list`
+                Ideally also has:
+                    - experiments (map)
+            metric_list: A list of columns that aggregate and compute
+                metrics, e.g.
+                `[F.coalesce(F.sum(df.metric_name), F.lit(0)).alias('metric_name')]`
+            last_date_full_data (str): The most recent date for which we
+                have complete data, e.g. '20190322'. If you want to ignore
+                all data collected after a certain date (e.g. when the
+                experiment recipe was deactivated), then do that here.
+            conv_window_start_days (int): the start of the conversion window,
+                measured in 'days since the user enrolled'. We ignore data
+                collected outside this conversion window.
+            conv_window_length_days (int): the length of the conversion window,
+                measured in days.
+            keep_client_id (bool): Whether to return a `client_id` column.
+                Defaults to False to reduce memory usage of the results.
+
+        Returns:
+            A spark DataFrame of experiment data. One row per `client_id`.
+            One or two metadata columns, then one column per metric in
+            `metric_list`. Then one column per sanity-check metric.
+            Columns:
+                - client_id (str, optional): Not necessary for
+                    "happy path" analyses.
+                - branch (str): The client's branch
+                - [metric 1]: The client's value for the first metric in
+                    `metric_list`.
+                - ...
+                - [metric n]: The client's value for the nth (final)
+                    metric in `metric_list`.
+                - [sanity check 1]: The client's value for the first
+                    sanity check metric.
+                - ...
+                - [sanity check n]: The client's value for the last
+                    sanity check metric.
+
+            This format - the schema plus there being one row per
+            enrolled client, regardless of whether the client has data
+            in `df` - was agreed upon by the DS team, and is the
+            standard format for queried experimental data.
+        """
+        req_dates_of_data = conv_window_start_days + conv_window_length_days
+
+        if self.num_dates_enrollment is None:
+            self._print_enrollment_window(last_date_full_data, req_dates_of_data)
+
+        # TODO accuracy: can/should we switch from submission_date_s3 to when the
```

Comment:
We made a decision re: activity vs submission date in [Bug 1422892](https://bugzilla.mozilla.org/show_bug.cgi?id=1422892). In short, we decided we should not trust activity date for business critical decisions.

If we want to reopen that discussion let's open a bug.

Reply:
Awesome, thanks for the reference. I'll read up on it. Probably nothing's changed since that decision - maybe the sharper focus on conversion windows has some effect on the weights of things being traded off?

Reply:
Maybe, but I'm concerned this comment will get folks hyper-focused on the activity date vs submission date difference when they're trying to debug weird experiment results.

Reply:
Under the rug it goes!